packages/mui-lab/src/internal/pickers/PureDateInput.tsx

```diff
@@ -145,7 +145,7 @@ export const PureDateInput = React.forwardRef(function PureDateInput(
       'aria-readonly': true,
       'aria-label': getOpenDialogAriaText(rawValue, utils),
       value: inputValue,
-      onClick: onOpen,
+      onClick: !props.readOnly ? onOpen : () => {},
```

Comment:
```suggestion
      ...(!props.readOnly && { onClick: onOpen }),
```

No need to set `onClick` at all if `readOnly` is set.

Reply:
Oh yes! Pushed the change. Thanks for the comment